### PR TITLE
[concurrency] conform Deque.Iterator to unchecked-Sendable

### PR DESCRIPTION
**Pull request by glessard**
On a toolchain that contains https://github.com/swiftlang/swift/pull/75782 (79bf5d51ca4b55afdb29102f411898d78cbe0bf2), `Deque` no longer compiles in Swift 6 mode.

`extension Deque._Storage: @unchecked Sendable {}` makes it compile, though it may not be the best solution?

Addresses rdar://134300306

### Checklist
- [x] I've read the [Contribution Guidelines](/README.md#contributing-to-swift-collections)
- [x] My contributions are licensed under the [Swift license](/LICENSE.txt).
- [x] I've followed the coding style of the rest of the project.
- [ ] I've added tests covering all new code paths my change adds to the project (if appropriate).
- [ ] I've added benchmarks covering new functionality (if appropriate).
- [ ] I've verified that my change does not break any existing tests or introduce unexplained benchmark regressions.
- [ ] I've updated the documentation if necessary.

### CODE REVIEW

**Comment by glessard** [comment]
@swift-ci please test

File: Sources/DequeModule/Deque._Storage.swift [hunk 1]
```diff
@@ -27,6 +27,8 @@ extension Deque {
   }
 }
 
+extension Deque._Storage: @unchecked Sendable {}
```
Comment (lorentey) [inline Sources/DequeModule/Deque._Storage.swift:30]: Does it make sense to declare `Deque._Storage` sendable, rather than only `Deque` itself?

Comment (glessard) [reply Sources/DequeModule/Deque._Storage.swift:30]: There already is a Sendable conformance for `Deque`. I'm not clear why the compiler is tripping over `_Storage`'s sendability status, but setting it separately fixes the error. It should, however, depend on the `Element` type.

**Comment by glessard** [comment]
@swift-ci please test

File: Sources/DequeModule/Deque._Storage.swift (hunk 1 above)
Comment (lorentey) [reply Sources/DequeModule/Deque._Storage.swift]: What I mean is that `Deque`'s conformance ought to be `@unchecked`.

**Review by lorentey — APPROVED** [review]

File: Sources/DequeModule/Deque._Storage.swift (hunk 1 above)
Comment (glessard) [reply Sources/DequeModule/Deque._Storage.swift]: Made `Deque.Iterator`'s conformance unchecked-Sendable; thanks for the nudge.

**Comment by glessard** [comment]
@swift-ci please test

**Review by lorentey — APPROVED** [review]

**Comment by glessard** [comment]
@swift-ci please test

**Pull request merged**